test/jdk/java/lang/ref/ReferenceClone.java

```diff
@@ -47,7 +47,9 @@ public void test() {
         CloneableReference ref = new CloneableReference(o);
         try {
             ref.clone();
-        } catch (CloneNotSupportedException e) {}
+        } catch (CloneNotSupportedException e) {
+            throw new RuntimeException("CloneableReference::clone should not throw CloneNotSupportedException");
+        }
```

Comment:
Alternatively, it could simply let CNSE propagate.

```
CloneableReference ref = new CloneableReference(o);
ref.clone();
```

`test()` and `main` will need to declare this checked exception.

Reply:
That was my initial thought, but it doesn't work - CNSE is a checked exception so must be handled.

Reply:
> test() and main will need to declare this checked exception.

```
diff --git a/test/jdk/java/lang/ref/ReferenceClone.java b/test/jdk/java/lang/ref/ReferenceClone.java
index bd1ead81bec..2f9386b81e4 100644
--- a/test/jdk/java/lang/ref/ReferenceClone.java
+++ b/test/jdk/java/lang/ref/ReferenceClone.java
@@ -31,12 +31,12 @@ import java.lang.ref.*;
 
 public class ReferenceClone {
     private static final ReferenceQueue<Object> QUEUE = new ReferenceQueue<>();
-    public static void main(String... args) {
+    public static void main(String... args) throws Exception {
         ReferenceClone refClone = new ReferenceClone();
         refClone.test();
     }
 
-    public void test() {
+    public void test() throws CloneNotSupportedException {
         // test Reference::clone that throws CNSE
         Object o = new Object();
         assertCloneNotSupported(new SoftRef(o));
@@ -45,9 +45,7 @@ public class ReferenceClone {
 
         // Reference subclass may override the clone method
         CloneableReference ref = new CloneableReference(o);
-        try {
             ref.clone();
-        } catch (CloneNotSupportedException e) {}
     }
 
     private void assertCloneNotSupported(CloneableRef ref) {
     ```

Reply:
Yes, that would work.  But I'd rather keep the code for that subtest close together, i.e. as currently written.